controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go

```diff
@@ -306,36 +303,33 @@ func (r *KubeadmControlPlaneReconciler) updateStatus(ctx context.Context, kcp *c
 	kcp.Status.UpdatedReplicas = int32(len(currentMachines))
 
 	replicas := int32(len(ownedMachines))
+
+	// set some basics in case the workload cluster is not yet available
 	kcp.Status.Replicas = replicas
-	readyMachines := int32(0)
+	kcp.Status.UnavailableReplicas = replicas
```

Comment:
I think this is wrong and should not be set if we can't contact the workload cluster